code/modules/mob/living/carbon/human/species_types/plasmamen.dm

```diff
@@ -51,7 +51,12 @@
 
 /datum/species/plasmaman/spec_life(mob/living/carbon/human/H, delta_time, times_fired)
 	var/atmos_sealed = CanIgniteMob(H) && (isclothing(H.wear_suit) && H.wear_suit.clothing_flags & STOPSPRESSUREDAMAGE) && (isclothing(H.head) && H.head.clothing_flags & STOPSPRESSUREDAMAGE)
-	if(!atmos_sealed && (!istype(H.w_uniform, /obj/item/clothing/under/plasmaman) || !istype(H.head, /obj/item/clothing/head/helmet/space/plasmaman) || !istype(H.gloves, /obj/item/clothing/gloves)))
+	var/list/flammable_limbs = list() //If any limbs are found in this list the plasmaman will attempt to autoignite
+	for(var/any_bodypart in H.bodyparts)
+		var/obj/item/bodypart/found_bodypart = any_bodypart
+		if(found_bodypart.status == BODYPART_ORGANIC && (found_bodypart.species_id == SPECIES_PLASMAMAN || HAS_TRAIT(found_bodypart, TRAIT_PLASMABURNT))) //Allows for "donated" limbs and augmented limbs to prevent autoignition
+			flammable_limbs += found_bodypart
```

Comment:
I think you'd be better off having this as a boolean and early returning because it'll cut down on processing times for plasmamen who are unmodified as it won't have to go through every bodypart.